src/lib.rs

```diff
@@ -85,3 +95,24 @@
     unused_results,
     variant_size_differences
 )]
+
+pub mod ast;
+mod combinator;
+#[cfg(feature = "expand-into-regex")]
+pub mod expand;
+pub mod parse;
+
+// TODO: Remove once `derive_more` 0.99.17 is released.
+use syn as _;
+
+#[doc(inline)]
+pub use self::ast::{
+    Alternation, Alternative, Expression, Optional, Parameter,
+    SingleAlternation, SingleExpression, Spanned,
+};
+#[cfg(feature = "expand-into-regex")]
+#[doc(inline)]
+pub use self::expand::{
+    ExpansionError as Error, IntoRegexCharIter, ParametersProvider,
```

Comment:
I think we can omit re-exporting feature in root. They're quite specific to live in their on module.